### PR TITLE
Setup grpc and grpc-gateway server in admin

### PR DESCRIPTION
**Pull request by rakeshsharma14317**

### CODE REVIEW

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: admin/cmd/main.go [hunk 1]
```diff
@@ -29,7 +32,7 @@ type Config struct {
 
 func main() {
 	// Load .env (note: fails silently if .env has errors)
-	_ = godotenv.Load()
+	_ = godotenv.Load("../.env")
```
[the remark this answers is not in the document]
Comment (rakeshsharma14317) [reply admin/cmd/main.go:35]: No i checked its not working, might be taking current dir, so i added `"../.env"` from current dir

File: admin/server/auth.go [hunk 2]
```diff
@@ -61,121 +69,133 @@ func (a *Authenticator) VerifyIDToken(ctx context.Context, token *oauth2.Token)
...
+	sess, err := store.Get(req, authSessionName)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+		http.Error(w, fmt.Sprintf("failed to get session: %s", err), http.StatusInternalServerError)
+		return
 	}
 
-	if c.QueryParam("state") != sess.Values["state"] {
-		return c.String(http.StatusBadRequest, "Invalid state parameter.")
+	if req.FormValue("state") != sess.Values["state"] {
```
[the remark this answers is not in the document]
Comment (rakeshsharma14317) [reply admin/server/auth.go:102]: Getting both the way,  we can change to `req.URL.Query().Get("state")` also

[the remark this answers is not in the document]
Comment (rakeshsharma14317) [reply admin/server/auth.go:102]: echo.Context has `QueryParam` method,

File: admin/server/auth.go [hunk 3]
```diff
@@ -61,121 +69,133 @@ func (a *Authenticator) VerifyIDToken(ctx context.Context, token *oauth2.Token)
...
+		http.Error(w, fmt.Sprintf("Invalid state parameter: %s", err), http.StatusBadRequest)
+		return
 	}
 
 	// Exchange an authorization code for a token.
-	token, err := s.auth.Exchange(c.Request().Context(), c.QueryParam("code"))
+	token, err := s.auth.Exchange(req.Context(), req.FormValue("code"))
```
[the remark this answers is not in the document]
Comment (rakeshsharma14317) [reply admin/server/auth.go:108]: same can be updated as `req.URL.Query().Get("code")`

[the remark this answers is not in the document]
Comment (rakeshsharma14317) [reply admin/server/auth.go:108]: what i found was it also returns the first value for the named component of the query. but yeah it for some method body parameters take precedence over URL query string values if we use FormValue

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**